File: Frontend/news-flow/src/models/News.ts


class News{
    id: number
    title: string
    imageUrl: string
    authorName: string
    summary: string
    text: string
    authorId: number
    viewsCount: number
    likeCount: number
    date: Date

    constructor(
        id: number = 0,
        title: string = "",
        url: string = "",
        authorName: string = "", 
        summary: string = "",
        text: string = "",
        authorId: number = 0,
        viewsCount: number = 0,
        likeCount: number = 0,
        date: Date = new Date())
    {
        this.id = id
        this.title = title
        this.imageUrl = url
        this.authorName = authorName
        this.summary = summary
        this.text = text
        this.authorId = authorId
        this.viewsCount = viewsCount
        this.likeCount = likeCount
        this.date = date
    }
    
}

export default News

File: Backend/NewsFlowAPI/Models/News.cs
﻿
namespace NewsFlowAPI.Models
{
    public class News
    {
        public long Id { get; set; }
        public string Title { get; set; } = String.Empty;
        public string Summary { get; set; } = String.Empty;

        public string Text { get; set; } = String.Empty;
        public string ImageUrl { get; set; } = String.Empty;

        public User? Author { get; set; }
        public long AuthorId { get; set; }

        public List<Tag> Tags { get; set; } = new List<Tag>();

        public Location? Location { get; set; }
        public long? LocationId { get; set; }
        public DateTime PostTime { get; set; }
        public int ViewsCount { get; set; }
        public int LikeCount { get; set; }
        public int ViewsLastPeriod { get; set; }
        public int LikesLastPeriod { get; set; }

        public DateTime LastPeriodTime { get; set; }
    }
}
